vision/objectdetection/color_detector.go

```diff
@@ -24,14 +26,23 @@ func NewColorDetector(cfg *ColorDetectorConfig) (Detector, error) {
 		return nil, err
 	}
 	hue, _, _ := col.HsvNormal()
-	tol := cfg.Tolerance
+	tol := cfg.HueTolerance
 	if tol > 1.0 || tol < 0.0 {
 		return nil, errors.Errorf("tolerance must be between 0.0 and 1.0. Got %.5f", tol)
 	}
```

Comment:
Since the Saturation and Value cutoffs are now included, can you put a check in NewColorDetector to see if the "detect_color" itself is fulfilled within the bounds selected? Like, if someone wants to find the color #000000, that will never be found if the Value cut-off is greater than 0. So return an error during construction of something like "detect_color has value/saturation X, which is out of bounds Y" would be good.

Reply:
Added.